### PR TITLE
Feature/#153 sample agent

### LINKED ISSUE #153: Move sample agents outside dialoguekit to a separate top-level sample_agents module

### PR DESCRIPTION
**Pull request by NoB0**
Fix #153 
- Move sample agents
- Make some methods from Participant and Agent abstract
- Update example usage in README
- Add test for `ParrotAgent`

### CODE REVIEW

**Comment by adrzewiecki** [comment]
In the pre-commit change:
`https://gitlab.com/pycqa/flake8` -> `https://github.com/pycqa/flake8`

This should fix the CI issue.

File: dialoguekit/participant/participant.py
Comment (kbalog) [inline dialoguekit/participant/participant.py:64]: Nit: decapitalize participant

File: dialoguekit/participant/participant.py
Comment (kbalog) [inline dialoguekit/participant/participant.py:59]: Should we not assume simply an Utterance here? For certain participants it might be an AnnotatedUtterance, but that is enough to restrict in the derived class.

File: sample_agents/__init__.py
Comment (kbalog) [inline sample_agents/__init__.py:2]: Add an issue and a TODO here for removing MathAgent

**Review by kbalog — CHANGES_REQUESTED** [review]
There is a question that may be addressed in a separate PR, but nevertheless would like to resolve before merging this one.

File: dialoguekit/participant/participant.py
Comment (NoB0) [reply dialoguekit/participant/participant.py:59]: Yes, I changed `AnnotatedUtterance` by `Utterance`. The derived class cannot use the type `AnnotatedUtterance ` without violating the Liskov substitution principle (see explanation [here](https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides)).

File: dialoguekit/participant/participant.py
Comment (NoB0) [reply dialoguekit/participant/participant.py:64]: Done.

File: sample_agents/__init__.py
Comment (NoB0) [reply sample_agents/__init__.py:2]: Should the issue include the removal of the class MathUser too?

Comment (kbalog) [reply sample_agents/__init__.py:2]: Yes please.

File: dialoguekit/participant/participant.py
Comment (kbalog) [reply dialoguekit/participant/participant.py:59]: There are workarounds -- we can cross that bridge when we get there. But I'm a bit confused now: we are not using mypy for type checking, are we?

Comment (NoB0) [reply dialoguekit/participant/participant.py:59]: We are.

File: sample_agents/__init__.py
Comment (NoB0) [reply sample_agents/__init__.py:2]: Issue #162

File: dialoguekit/participant/participant.py
Comment (kbalog) [reply dialoguekit/participant/participant.py]: Where does that come from? It is not part of our [default setup](https://github.com/iai-group/guidelines/tree/main/python) nor [project template](https://github.com/iai-group/template-project). I missed when it was added and I'd argue that we don't want it unless there are specific reasons for having it.

**Review by kbalog — APPROVED** [review]
LGTM, but there is a pending question about the use of mypy. Please create a separate issue about removing it (or to have a discussion about it there).

File: dialoguekit/participant/participant.py
Comment (NoB0) [reply dialoguekit/participant/participant.py]: Resolved internally.

**Pull request merged**
